main/internal/implementations/dragon_drop/http_dragondrop_managed_visualization.go

```diff
@@ -1,3 +1,408 @@
+	}, newResources, nil
+}
+
+// getCloudCostsData returns the costs of the resources outside of terraform control and the costs of the resources
+func (c *HTTPDragonDropClient) getCloudCostsData(ctx context.Context, newResources map[string]interface{}) (CloudCostsData, error) {
+	costEstimation, err := readOutputFileAsSlice("cost-estimates.json")
+	if err != nil {
+		return CloudCostsData{}, err
+	}
+
+	if len(costEstimation) == 0 {
+		return CloudCostsData{}, nil
+	}
+
+	cloudCostsData := CloudCostsData{}
+	for _, costEstimate := range costEstimation {
+		costEstimateMap := costEstimate.(map[string]interface{})
+		_, ok := newResources[costEstimateMap["resource_name"].(string)]
+
+		if ok {
+			cloudCostsData.CostsOutsideOfTerraform += getMonthlyCost(costEstimateMap)
+			cloudCostsData.CostsOutsideOfTerraform = roundFloat(cloudCostsData.CostsOutsideOfTerraform)
+		} else {
+			cloudCostsData.CostsTerraformControlled += getMonthlyCost(costEstimateMap)
+			cloudCostsData.CostsTerraformControlled = roundFloat(cloudCostsData.CostsTerraformControlled)
+		}
+	}
+
+	return cloudCostsData, nil
+}
+
+// getMonthlyCost returns the monthly cost of a resource
+func getMonthlyCost(costEstimateMap map[string]interface{}) float64 {
+	monthlyCost := costEstimateMap["monthly_cost"].(string)
+	if strings.Trim(monthlyCost, "") == "" || monthlyCost == "0" {
+		price := costEstimateMap["price"].(string)
+		priceFloat, err := strconv.ParseFloat(price, 64)
+		if err != nil {
+			return 0
+		}
+
+		return priceFloat
+	}
+
+	monthlyCostFloat, err := strconv.ParseFloat(monthlyCost, 64)
+	if err != nil {
+		return 0
+	}
+
+	return monthlyCostFloat
+}
+
+// roundFloat rounds a float to 3 decimal places
+func roundFloat(val float64) float64 {
+	ratio := math.Pow(10, float64(3))
+	return math.Round(val*ratio) / ratio
+}
+
+// getCloudSecurityData returns the number of security risks found in the security scan
+func (c *HTTPDragonDropClient) getCloudSecurityData(ctx context.Context) (CloudSecurityData, error) {
+	securityScan, err := readOutputFileAsMap("security-scan.json")
+	if err != nil {
+		return CloudSecurityData{}, err
+	}
+
+	cloudSecurityData := CloudSecurityData{}
+	results := securityScan["results"].([]interface{})
+	for _, result := range results {
+		severity := result.(map[string]interface{})["severity"].(string)
+		if severity == "CRITICAL" {
+			cloudSecurityData.SecurityRiskCritical++
+		} else if severity == "HIGH" {
+			cloudSecurityData.SecurityRiskHigh++
+		} else if severity == "MEDIUM" {
+			cloudSecurityData.SecurityRiskMedium++
+		} else if severity == "LOW" {
+			cloudSecurityData.SecurityRiskLow++
+		}
+	}
+
+	return cloudSecurityData, nil
+}
+
+// getTerraformFootprintData returns terraform footprint data for all terraform files
+func (c *HTTPDragonDropClient) getTerraformFootprintData(ctx context.Context) (TerraformFootprintData, error) {
+	files := []string{"versions.tf", "main.tf"}
+	files = append(files, getAllTFFiles(ctx, "current_cloud")...)
```

Comment:
I'm not sure if this is the case, but are we at risk of "double counting" the values we find in `versions.tf` and `main.tf` files?

Reply:
I think the searching here needs to be implemented a bit differently:
We need to search through all of the user's files within each of the specified workspace directories in their repo, not the current-cloud definition (that is generated by Terraformer).